ddtrace/tracer/post_processor.go

```diff
@@ -0,0 +1,111 @@
+package tracer
+
+import (
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/internal"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+// Note: Keeping everything in this file for easy access during development.
+// I will likely move the ReadWriteSpan interface to the ddtrace package,
+// and the (s *span) *() methods to span.go.
+
+// ReadOnlySpan specifies methods to read from a span.
+type readOnlySpan interface {
+	// Span name returns the operation of the span.
+	SpanName() string
+	// etc...
+}
+
+// ReadWriteSpan implements the methods of ddtrace.Span (to write to a span)
+// and readOnlyspan (to read from a span).
+type ReadWriteSpan interface {
+	ddtrace.Span
+	readOnlySpan
+	Remove()
+}
+
+// Span name returns the operation name of s.
+func (s *span) SpanName() string {
+	s.Lock()
+	defer s.Unlock()
+	// the if statement below is not necessary to
+	// prevent a race condition with spans being
+	// flushed, as this is a read. I decided
+	// to still have this because it should be
+	// a user error to call this on a finished span
+	// (that is not being processed). I can remove it.
+	if s.finished && !s.inProcessor {
```

Comment:
Why wouldn't the name of the span be accessible at any time regardless of whether it is finished or not?

Reply:
You're right, removed this part 👍